### PR TITLE
Variant replaced with Runtime Attribute

### PR DESCRIPTION
**Pull request by apankratovantonp**
- Based on #8288 
- `ov::Variant` replaced with `ov::RuntimeAttribute`
- `ov::Any` extended to support access by parent class reference (if  `Base member type is defined`)
- Transformation library runtime parameters refactored to use direct inheritance from `ov::RuntimeAttribute`

### CODE REVIEW

File: inference-engine/src/offline_transformations/include/mask_attribute.hpp [hunk 1]
```diff
@@ -28,6 +27,11 @@ namespace ngraph {
 class Mask : public std::vector<std::set<uint64_t>>,
              public std::enable_shared_from_this<Mask> {
 public:
+    static const ::ov::DiscreteTypeInfo& get_type_info_static() {
```
Comment (ilyachur) [inline inference-engine/src/offline_transformations/include/mask_attribute.hpp:30]: Can we use OPENVINO_RTTI?

Comment (apankratovantonp) [reply inference-engine/src/offline_transformations/include/mask_attribute.hpp:30]: if I am not mistaken `OPENVINO_RTTI` can be used if have already  `virtual get_type_info()` in the class hierarchy. This class does not inherit something.

File: docs/snippets/InferenceEngine_Caching3.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@ using namespace InferenceEngine;
     auto it = std::find(keys.begin(), keys.end(), METRIC_KEY(IMPORT_EXPORT_SUPPORT));
 
     // If metric 'IMPORT_EXPORT_SUPPORT' exists, check it's value
-    bool cachingSupported = (it != keys.end()) && ie.GetMetric(deviceName, METRIC_KEY(IMPORT_EXPORT_SUPPORT)).as<bool>();
+    auto cachingSupported = (it != keys.end()) && ie.GetMetric(deviceName, METRIC_KEY(IMPORT_EXPORT_SUPPORT)).as<bool>();
```
[the remark this answers is not in the document]
Comment (apankratovantonp) [reply docs/snippets/InferenceEngine_Caching3.cpp:17]: As bool type will be declared in `as<bool>()` method.

File: src/common/transformations/include/transformations/fix_rt_info.hpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+class NGRAPH_API FixRtInfo;
+
+}  // namespace pass
+}  // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief FixRtInfo transformation helps to set fix info attributes in a single place.
+ *        User can pass runtime attribute using various wais.
```
Comment (ilya-lavrenov) [inline src/common/transformations/include/transformations/fix_rt_info.hpp:35]: `ways`

Comment (ilya-lavrenov) [reply src/common/transformations/include/transformations/fix_rt_info.hpp:35]: CC @GlebKazantaev

File: src/core/include/openvino/core/any.hpp [hunk 4]
```diff
@@ -264,11 +337,12 @@ class OPENVINO_API Any {
...
     void impl_check() const;
 
-    mutable Base::Ptr _variant_impl;
-
-    Base::Ptr _impl;
+    mutable Base::Ptr _runtime_attribute_impl;
 
 public:
+    Base::Ptr
+        _impl;  ///< Pointer to implementation. TODO: make it private when ngraph::RuntimeAttributeWrapper is deprecated
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/any.hpp:344]: when `RuntimeAttributeWrapper ` will be deprecated?

File: src/core/include/openvino/core/runtime_attribute.hpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    RuntimeAttributeWrapper(const value_type& value) : RuntimeAttributeImpl<value_type>(value) {}
+    std::string to_string() const override {
+        return std::to_string(m_value);
+    }
+};
+
+template <typename T>
+inline std::shared_ptr<RuntimeAttribute> make_runtime_attribute(const T& p) {
+    return std::static_pointer_cast<RuntimeAttribute>(std::make_shared<RuntimeAttributeWrapper<T>>(p));
+}
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/runtime_attribute.hpp:97]: @GlebKazantaev do we need such helpers?

File: src/core/include/openvino/core/runtime_attribute.hpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+public:
+    OPENVINO_RTTI("RuntimeAttributeWrapper<std::string>");
+    RuntimeAttributeWrapper(const value_type& value) : RuntimeAttributeImpl<value_type>(value) {}
+    std::string to_string() const override {
+        return m_value;
+    }
+};
+
+template <>
+class OPENVINO_API RuntimeAttributeWrapper<int64_t> : public RuntimeAttributeImpl<int64_t> {
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/runtime_attribute.hpp:85]: why do we need `RuntimeAttributeWrapper`? for simple types?

**Review by ilya-lavrenov — APPROVED** [review]

File: src/core/include/openvino/core/runtime_attribute.hpp (hunk 6 above)
Comment (apankratovantonp) [reply src/core/include/openvino/core/runtime_attribute.hpp:85]: Will be removed in separate PR https://github.com/openvinotoolkit/openvino/pull/8946

File: src/core/include/openvino/core/any.hpp (hunk 4 above)
Comment (apankratovantonp) [reply src/core/include/openvino/core/any.hpp]: moved to private

File: src/core/include/openvino/core/runtime_attribute.hpp (hunk 5 above)
Comment (apankratovantonp) [reply src/core/include/openvino/core/runtime_attribute.hpp:97]: also will be removed in PR #8946

File: src/common/transformations/include/transformations/fix_rt_info.hpp (hunk 3 above)
Comment (GlebKazantaev) [reply src/common/transformations/include/transformations/fix_rt_info.hpp]: @apankratovantonp I don't think that we need dedicated pass for rt info fix as it duplicates graph traversals inside InitNodeInfo. Could you please move all this logic from fix_rt_info.hpp to init_node_info.cpp?

Comment (GlebKazantaev) [reply src/common/transformations/include/transformations/fix_rt_info.hpp]: can be done in a separate PR.

**Review by GlebKazantaev — APPROVED** [review]

File: src/common/transformations/include/transformations/fix_rt_info.hpp (hunk 3 above)
Comment (apankratovantonp) [reply src/common/transformations/include/transformations/fix_rt_info.hpp]: This pass is not the same as the init node info! It does not change rt info if it is empty!

**Pull request merged**